torch/_higher_order_ops/wrap.py

```diff
@@ -140,6 +141,11 @@ def __call__(self, gmod, *args, **kwargs):
             # And we ensure that AOT Autograd traces through the non reentrant
             # version of checkpointing.
             kwargs["use_reentrant"] = False
+            # preserve_rng_state is set to False because we want to prevent AOTAutograd from tracing through
+            # `torch.random.fork_rng` op (which is not supported yet under CUDA).
+            # This doesn't mean that we don't preserve RNG state. Instead, we will always preserve RNG state
+            # (i.e. do RNG functionalization via `replace_random_passes` in Inductor) regardless of this flag.
```

Comment:
is this always true? I am looking at the https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/replace_random.py#L94, and it seems to me that it would replace the randn ops with inductor prim rand, but the seed are always newly created seed?

Reply:
the seed is newly created at compiled time, and it's then hardcoded into the graph. So at runtime we always reuse the same seed for the same random op.

Reply:
hmm isn't RNG functionalization controlled by this flag? Which looks like it's off by default - do we assert anywhere that it must be set when those code runs? https://github.com/pytorch/pytorch/blob/main/torch/_functorch/config.py#L16

Reply:
It's complicated: AOTAutograd RNG functionalization is controlled by this flag that you mentioned (https://github.com/pytorch/pytorch/blob/b526aae95a7a9090ba7b628ce36d6e5d254a171e/torch/_functorch/config.py#L16), but Inductor RNG functionalization is done here https://github.com/pytorch/pytorch/blob/b526aae95a7a9090ba7b628ce36d6e5d254a171e/torch/_inductor/fx_passes/joint_graph.py#L277 and not controlled by that flag.

For both AC and SAC, we only use Inductor RNG functionalization, unless the functorch functionalize_rng_ops flag is on. (It is decided here: https://github.com/pytorch/pytorch/blob/b526aae95a7a9090ba7b628ce36d6e5d254a171e/torch/_dynamo/utils.py#L1976-L1980)